clients/src/main/resources/common/message/ConsumerGroupDescribeResponse.json

```diff
@@ -69,7 +70,9 @@
             { "name": "Assignment", "type": "Assignment", "versions": "0+",
               "about": "The current assignment." },
             { "name": "TargetAssignment", "type": "Assignment", "versions": "0+",
-              "about": "The target assignment." }
+              "about": "The target assignment." },
+            { "name": "IsClassic", "type": "bool", "versions": "1+", "ignorable": true,
+              "about": "True for classic member." }
```

Comment:
In the RPC layer, everything should function correctly since the new broker does not return the `IsClassic` field for version 0.

The primary issue lies in how the Java client distinguishes between "false" and "unknown" values, as it always defaults to "false." To address this, we could add a `ConsumerGroupDescribeResponse#version` method that returns the response version. This would allow the caller to differentiate between "false" and "unknown" based on the version.

Additionally, the `MemberDescription#isClassic` field also needs to differentiate between "false" and "unknown." One possible solution is to use `Optional<Boolean>` instead.

Reply:
Hi @dajac / @chia7712, thanks for the review. It looks like we can implement it in Admin side. If `ConsumerGroupDescribeResponse#version` is `0`, set `MemberDescription#isClassic` as `Optional.empty()`. If `ConsumerGroupDescribeResponse#version` is `1`, set `MemberDescription#isClassic` as `Optional.of(value)`. I change the KIP to align this discussion.

Reply:
Using an Optional makes total sense. For the RPC, I was wondering whether we want to make it more explicit in the schema. For instance, we could use a int8 with three values: -1, 0, 1. -1 would be used by default and would mean no provided. Relying on the version also works but it may be easy to miss for people implementing other clients.

Reply:
I personally prefer to have a bit more complexity on the server rather than relying on all the developers of clients to do the right thing.

Reply:
> I personally prefer to have a bit more complexity on the server rather than relying on all the developers of clients to do the right thing.

I tried to imagine the code for both schemas.

### int8

client code
```java
switch (data.isClassic()) {
  case 0: return Optional.of(false);
  case 1: return Optional.of(true);
  default: return Optional.empty();
}
```
server code
```java
data.setIsClassic(useClassicProtocol ? 1 : 0);
```
---
### bool

client code
```java
if (version < 1) return Optional.empty();
return Optional.of(data.isClassic());
```

server
```java
data.setIsClassic(useClassicProtocol);
```
---

if I'm imagining correctly, it seems that using a bool could simplify the code.

Reply:
Given that the response data may be propagated without version information, using the data itself to distinguish "undefined" values is safer. Additionally, I support using `int8`.